### DISCUSSION TITLE
Knightsbridge / Greencycle 5-gang IP66 power switch - rolling codes?

### DISCUSSION THREAD

**Discussion by ewanda** [arendst/Tasmota#11233] [Q&A] [answered]
hi - my garden landscaper has installed a 5-gang water resistant power switch to control garden lights, and it comes with a 433Mhz RF remote with 10 buttons, to enable on & off for 5 different zones. The remote is paired with the switch box through a single button on the circuit board of the box; you then press any of the buttons on the remote and it just works.

![image](https://user-images.githubusercontent.com/39132799/110157045-a4250280-7ddf-11eb-92df-583dc9ba9023.png)

In the UK, it's sold as a Knightsbrige IP665G, but it's also been available under the name Greencycle in the US previously. It's Chinese made but I haven't managed to track down the actual manufacturer and there's no obvious equivalent product under other names that I can find.

Now, I've tried automating it through a variety of means - I have a Sonoff RF-Wifi gateway which is flashed with latest releases of Tasmota and Portisch; the procedure for learning the sequence and then sending it back to a remote device has been tested with a simple fixed code power switch and it worked first time, but going via **rfraw 177** on the remote for the switch box gives me nothing, regardless of how many samples I've tried.

I can only imagine it is using a rolling code and is unable to reliably work with the Sonoff - does anyone have any suggestions of other things to try? There's no obvious chip identification, though the back of the circuit board in the switch has a 14-pin IC and the remote has one that looks very much the same (the match button is in the very top top left on the opposite side):

![image](https://user-images.githubusercontent.com/39132799/110157913-a50a6400-7de0-11eb-8852-39130cc9c105.png)

many thanks - I'm tearing my hair out!
Ewan

**Comment by lalo-uy** [comment, marked as answer]
I bet you can check the codes been transmitted in the test pad between C15
and R17. Try to see is the same button sends different codes.

**Comment by ewanda** [comment]
hi - thanks for the suggestion, though I'm not sure on what you're saying - are you talking about connecting a probe to that pad on the circuit board to measure the signal? That pad is on the data out line from the 8-pin RF chip (480R 2010) but I haven't managed to figure out what the 14-pin decoder IC is.

When I put the Sonoff box into RFRAW 177 and press a button on the remote repeatedly, I'll get the likes of

{"RfRaw":{"Data":"AA   B1 03 01EA 03F2 03A2 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01E0 03F2 03AC 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01EA 03E8 03AC 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01EA 03D4 038E 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01F4 03DE 0398 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01F4 03D4 038E 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01EA 03DE 03A2 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01F4 03DE 03A2 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01FE 03CA 0398 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01D6 03F2 03AC 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01CC 03FC 0348 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01E0 03F2 03AC 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01CC 03F2 03B6 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01EA 03DE 03B6 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01E0 03F2 0398 28180818180808 55"}}
{"RfRaw":{"Data":"AA   B1 03 01F4 03CA 03B6 28180818180808 55"}}

... but converting those to B0 values using https://bbconv.hrbl.pl/ and then sending back ( eg 
cmnd/sonoffGW/Backlog   RfRaw AAB00F030801E003F203AC2818081818080855)

... does nothing. So I think I have what is being sent by the remote OK, but I'm presuming that it's using a rolling code so I;m a bit stuck.
